### PR TITLE
Add server action methods

### PR DESCRIPTION
**Pull request by pyama86**
Hi!!
Ucchi→♡♥♡
I think I have a cold. Take care not to catch a cold.

```ruby
require 'yao'                                                                                                                                                                                     
Yao.configure do
  auth_url "#{ENV["OS_AUTH_URL"]}/tokens"
  username ENV["OS_USERNAME"]
  password ENV["OS_PASSWORD"]
  tenant_name 'example'
end

Yao::Server.shutoff("xx_server_id_xx") if Yao::Server.get("xx_server_id_xx").status == "ACTIVE"
```

### CODE REVIEW

File: lib/yao/resources/action.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Yao::Resources
+  module Action
+    def action(id, query)
+      res = POST(action_path(id)) do |req|
+        req.body = query.to_json
+        req.headers['Content-Type'] = 'application/json'
```
Comment (udzura) [inline lib/yao/resources/action.rb]: I think yao's client automaticallt adds these request headers.
https://github.com/yaocloud/yao/blob/master/lib/yao/client.rb#L48
Is it required?

**Comment by udzura** [comment]
I don't know who Ucchi is, but I will take care of my health.

File: lib/yao/resources/action.rb (hunk 1 above)
Comment (pyama86) [inline lib/yao/resources/action.rb]: yes it is.
comment out this part
`req.headers['Content-Type'] = 'application/json'`
* request debug
```
#<Faraday::Env 
	@method=:post 
	@body="{\"os-stop\":null}" 
	@url=#<URI::HTTP http://example.api.com:8774/v2/b598bf98671c47e1b955f8c9660e3c44/servers/cee879b6-aefd-4a0d-8f61-85a5b0ecaec5/action> 
	@request=#<Faraday::RequestOptions (empty)> 
	@request_headers={"User-Agent"=>"Faraday v0.9.2", "Content-Type"=>"application/x-www-form-urlencoded", "X-Auth-Token"=>"3c28a51d13624c45827785c9a35ba7f3"} 
	@ssl=#<Faraday::SSLOptions (empty)>
>
```
json is not included in the request_headers.

Comment (udzura) [inline lib/yao/resources/action.rb]: Wow, I found `c.request :json` won't add any Content-Type headers:
https://github.com/lostisland/faraday_middleware/blob/master/lib/faraday_middleware/request/encode_json.rb

So this header should be explicit!
I understand.

**Comment by udzura** [comment]
LGTM!

**Pull request merged**